### PR TITLE
[New Rule] [BBR] Memory Dump File Rules

### PR DESCRIPTION
**Pull request by w0rk3r**
## Issues

https://github.com/elastic/ia-trade-team/issues/166

## Summary

Adds 2 Rules:

* Memory Dump File with Unusual Extension
  * 2 hits on RS (last 1y), one `.dump` extension, and a no-extension dropbox dump
* Potential Credential Access via Memory Dump File Creation
  * Adapted from Credential Access via Memory Dump File Creation deprecated ER, 34 hits last 90d on RS, should be good as BBR

### CODE REVIEW

File: rules_building_block/credential_access_mdmp_file_creation.toml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+id = "T1003.001"
+name = "LSASS Memory"
+reference = "https://attack.mitre.org/techniques/T1003/001/"
+
+
+
+[rule.threat.tactic]
+id = "TA0006"
+name = "Credential Access"
+reference = "https://attack.mitre.org/tactics/TA0006/"
```
Comment (Aegrah) [inline rules_building_block/credential_access_mdmp_file_creation.toml:88]:
```suggestion
name = "Credential Access"
reference = "https://attack.mitre.org/tactics/TA0006/"

```

File: rules_building_block/credential_access_mdmp_file_unusual_extension.toml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+[[rule.threat.technique]]
+id = "T1036"
+name = "Masquerading"
+reference = "https://attack.mitre.org/techniques/T1036/"
+
+
+[rule.threat.tactic]
+id = "TA0005"
+name = "Defense Evasion"
+reference = "https://attack.mitre.org/tactics/TA0005/"
```
Comment (Aegrah) [inline rules_building_block/credential_access_mdmp_file_unusual_extension.toml:73]:
```suggestion
name = "Defense Evasion"
reference = "https://attack.mitre.org/tactics/TA0005/"

```

**Review by Aegrah — APPROVED** [review]
LGTM!

File: rules_building_block/credential_access_mdmp_file_creation.toml (hunk 1 above)
Comment (w0rk3r) [reply rules_building_block/credential_access_mdmp_file_creation.toml:88]: Same as https://github.com/elastic/detection-rules/pull/3131#discussion_r1340101958

**Review by shashank-elastic — APPROVED** [review]

**Pull request merged**